packages/modules-sdk/src/medusa-app.ts

```diff
@@ -108,14 +108,20 @@ async function loadModules(modulesConfig, injectedDependencies) {
   return allModules
 }
 
-async function initializeLinks(linkModules, injectedDependencies) {
+async function initializeLinks(config, linkModules, injectedDependencies) {
   try {
-    const { initialize: initializeLinks } = await import(LinkModulePackage)
-    await initializeLinks({}, linkModules, injectedDependencies)
-    return new RemoteLink()
+    const { initialize } = await import(LinkModulePackage)
+    const linkResolution = await initialize(
+      config,
+      linkModules,
+      injectedDependencies
+    )
+
+    return { remoteLink: new RemoteLink(), linkResolution }
   } catch (err) {
     console.warn("Error initializing link modules.", err)
-    return undefined
+
+    return { remoteLink: undefined, linkResolution: undefined }
```

Comment:
suggestion: If we don't succeed to initialize a link, then we can start in an instable state. should we throw instead and not allow to start the app?